Sample to demonstrate custom `MessageConverter` to handle converting …

gvensan opened the pull request:
Sample to demonstrate custom `MessageConverter` to handle converting null payloads to custom POJOs.

How to use:
- Run the cloud-stream-sink-null-payload-converter 
- Publish an empty message from TryMe with an empty payload on the topic 'sensor/temperature/fahrenheit', which will publish a message on the topic with a null payload
- See that the null message passes through the converter and is returned as an empty POJO

Hope this the expectation in terms of sample for supporting custom message converter.

gvensan committed:
Sample to demonstrate custom `MessageConverter` to handle converting null payloads to custom POJOs

gvensan committed:
Updated spring boot version to 2.6.4 to be compatible with spring cloud version 2021.0.1 (bom 2.3.0)

Mrc0113 commented on cloud-stream-sink-null-payload-converter/pom.xml:
```diff
@@ -0,0 +1,96 @@
+
+	<properties>
+		<spring-cloud.version>2021.0.1</spring-cloud.version>
+		<solace-spring-cloud-bom.version>2.3.0</solace-spring-cloud-bom.version>
```
please update with latest versions as Jeff requested in the JIRA.

Mrc0113 replied:
bonus to also update the versions of the other Spring Cloud Stream samples in this repo and make sure they still run (it should "just work" ;) )

Mrc0113 commented on cloud-stream-sink-null-payload-converter/src/main/java/com/solace/samples/spring/scs/TemperatureSink.java:
```diff
@@ -0,0 +1,54 @@
+		return System.out::println;
+	}
+	
+    @Bean
```
add a comment that says something along the lines of

> if the payload is null, then the "solace_scst_nullPayload" message header will be set to true. Because the Spring framework does not allow for a message to have a null payload we need to add a custom message converter to handle that scenario.

gvensan replied:
Done!

Mrc0113 commented on cloud-stream-sink-null-payload-converter/src/test/java/com/solace/samples/spring/scs/TemperatureSinkTest.java:
```diff
@@ -0,0 +1,50 @@
+		BeanFactoryChannelResolver channelResolver = context.getBean("integrationChannelResolver",
+				BeanFactoryChannelResolver.class);
+		MessageChannel channel = channelResolver.resolveDestination("sink-in-0");
+		channel.send(MessageBuilder.withPayload(new SensorReading("test", 50, BaseUnit.FAHRENHEIT)).build());
```
Testing this would be interesting since Spring won't let us send a msg with a null payload. so what I would recommend here is to test manually that it works, delete this file and throw a comment in the main java class with a command they could use to test this via curl or sdkperf....or if you can use try-me just state how to do that :)

gvensan replied:
I chose not to remove the unit test code. Instead, added the comment about null-payload in spring and approaches to generate such messages manually. Hope that's fine.

Mrc0113 reviewed (changes requested):
Please see individual comments in this review.

gvensan committed:
- Updated version info of spring cloud, spring boot and spring cloud bom on all samples.
- Added appropriate comments to the null-payload-converter sample and updated methods to generate events with null payload (as spring doesn't allow that)
